tests/catch/unit/stream/hipStreamGetPriority.cc

```diff
@@ -56,35 +50,43 @@ TEST_CASE("Unit_hipStreamGetPriority_happy") {
   HIP_CHECK(hipDeviceGetStreamPriorityRange(&priority_low, &priority_high));
   hipStream_t stream{};
   int priority = 0;
-  SECTION("Default Priority") {
-    HIP_CHECK(hipStreamCreate(&stream));
-    HIP_CHECK(hipStreamGetPriority(stream, &priority));
+  SECTION("Null Stream") {
+    HIP_CHECK(hipStreamGetPriority(nullptr, &priority));
     // valid priority
-    // Lower the value higher the priority, higher the value lower the priority
     REQUIRE(priority_low >= priority);
     REQUIRE(priority >= priority_high);
   }
-  SECTION("High Priority") {
-    HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamDefault, priority_high));
-    HIP_CHECK(hipStreamGetPriority(stream, &priority));
-    REQUIRE(priority == priority_high);
-  }
-  SECTION("Higher Priority") {
-    HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamNonBlocking, priority_high - 1));
-    HIP_CHECK(hipStreamGetPriority(stream, &priority));
-    REQUIRE(priority == priority_high);
+  SECTION("Created Stream") {
+    SECTION("Default Priority") {
+      HIP_CHECK(hipStreamCreate(&stream));
+      HIP_CHECK(hipStreamGetPriority(stream, &priority));
+      // valid priority
+      // Lower the value higher the priority, higher the value lower the priority
+      REQUIRE(priority_low >= priority);
+      REQUIRE(priority >= priority_high);
+    }
+    SECTION("High Priority") {
+      HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamDefault, priority_high));
+      HIP_CHECK(hipStreamGetPriority(stream, &priority));
+      REQUIRE(priority == priority_high);
+    }
+    SECTION("Higher Priority") {
+      HIP_CHECK(hipStreamCreateWithPriority(&stream, hipStreamNonBlocking, priority_high - 1));
```

Comment:
Not sure what is the expected behavior if highest priority is 0? Then priority_high - 1 would be equal to -1. I think this test might fail.

Reply:
the expected behavior is for the value to be clamped to the priority range, so the resulting stream will have a priority of 0 in your example. The nvidia machine i tested had `priority_high` of -5 and the amd one had `priority_high` of -1, so I think this should be fine.

Reply:
Now it works that way but I don't think priorities are defined as such. Someone can choose low=0=high so expecting -1 to be fine is not really fine.

Reply:
Sorry, I'm not sure I understand the problem.

For CUDA, specifying a priority outside the range of `priority_high` to `priority_low` will result in the priority of the created stream to be clamped to that range. This is documented here https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1gfdb79818f7c0ee7bc585648c91770275 in the description for both `cudaDeviceGetStreamPriorityRange` and `cudaStreamCreateWithPriority`.

I can find no documentation mentioning priority values should be greater than or equal to zero. All the apis seems to use a signed int value for priority and, as I mentioned, I have found the devices I have access to have a `priority_high` of less than 0.

The only issue I can foresee is if `priority_high` is equal to `std::numeric_limits<int>::min`, leading to underflow.

Reply:
OK I missed the clamping part. Thanks!